### DISCUSSION TITLE
opentelemetry uses TCP or UDP ?

### DISCUSSION THREAD

**Discussion by serj-maks** [open-telemetry/opentelemetry-java#6240] [Q&A] [answered]
hi! Which transport opentelemetry via http and opentelemetry via gRPC use: TCP or UDP? [Otlp http doc](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#endpoint-urls-for-otlphttp) says about TCP (this is only place, that talks about transport). Google says, what gRPC usually use TCP, but in otlp doc I couldn't find any info about it

**Comment by jack-berg** [comment, marked as answer]
Since the `grpc` is built on top of http/2, the `grpc` version of OTLP uses TCP. The `http/protobuf` version of OTLP uses http/2 or http/1.1 so it uses TCP as well.

**Reply by serj-maks** [reply]
thank you!
